app/upgrades/v05/upgrades.go

```diff
@@ -0,0 +1,236 @@
+		ctx := sdk.UnwrapSDKContext(c)
+		logger := ctx.Logger().With("upgrade", UpgradeName)
+
+		// === Params migration ===
+		// https://github.com/cosmos/cosmos-sdk/pull/12363/files
+		// Set param key table for params module migration
+		for _, subspace := range keepers.ParamsKeeper.GetSubspaces() {
+			subspace := subspace
+
+			var keyTable paramstypes.KeyTable
+			switch subspace.Name() {
+			// SDK
+			case authtypes.ModuleName:
+				keyTable = authtypes.ParamKeyTable() //nolint:staticcheck
+			case banktypes.ModuleName:
+				keyTable = banktypes.ParamKeyTable() //nolint:staticcheck
+			case crisistypes.ModuleName:
+				keyTable = crisistypes.ParamKeyTable() //nolint:staticcheck
+			case distrtypes.ModuleName:
+				keyTable = distrtypes.ParamKeyTable() //nolint:staticcheck
+			case govtypes.ModuleName:
+				keyTable = govv1.ParamKeyTable() //nolint:staticcheck
+			case minttypes.ModuleName:
+				keyTable = minttypes.ParamKeyTable() //nolint:staticcheck
+			case slashingtypes.ModuleName:
+				keyTable = slashingtypes.ParamKeyTable() //nolint:staticcheck
+			case stakingtypes.ModuleName:
+				keyTable = stakingtypes.ParamKeyTable() //nolint:staticcheck
+
+			// IBC
+			case ibctransfertypes.ModuleName:
+				keyTable = ibctransfertypes.ParamKeyTable() //nolint:staticcheck
+			case icahosttypes.SubModuleName:
+				keyTable = icahosttypes.ParamKeyTable() //nolint:staticcheck
+			case icacontrollertypes.SubModuleName:
+				keyTable = icacontrollertypes.ParamKeyTable() //nolint:staticcheck
+			case icqtypes.ModuleName:
+				keyTable = icqtypes.ParamKeyTable() //nolint:staticcheck
+
+			// Wasm
+			case wasmtypes.ModuleName:
+				keyTable = wasmv2.ParamKeyTable() //nolint:staticcheck
+
+			default:
+				continue
+			}
+			if !subspace.HasKeyTable() {
+				subspace.WithKeyTable(keyTable)
+			}
+		}
+		// Migrate Tendermint consensus parameters from x/params module to a deprecated x/consensus module.
+		// The old params module is required to still be imported in your app.go in order to handle this migration.
+		baseAppLegacySS := keepers.ParamsKeeper.Subspace(baseapp.Paramspace).WithKeyTable(paramstypes.ConsensusParamsKeyTable())
+		err := baseapp.MigrateParams(ctx, baseAppLegacySS, keepers.ConsensusParamsKeeper.ParamsStore)
+		if err != nil {
+			return nil, err
+		}
+
+		// === New params ===
+		// https://github.com/cosmos/ibc-go/blob/v7.1.0/docs/migrations/v7-to-v7_1.md
+		// explicitly update the IBC 02-client params, adding the localhost client type
+		var newIBCCoreParams clienttypes.Params
+		newIBCCoreParams.AllowedClients = append(newIBCCoreParams.AllowedClients,
+			ibcexported.Solomachine,
+			ibcexported.Tendermint,
+			ibcexported.Localhost,
+		)
+		// ICA Host
+		newIcaHostParams := icahosttypes.Params{
+			HostEnabled: true,
+			// https://github.com/cosmos/ibc-go/blob/v4.2.0/docs/apps/interchain-accounts/parameters.md#allowmessages
+			AllowMessages: []string{"*"},
```

Comment:
We don't need to set the Allow Messages directly not "*"

like below example
```suggestion
			AllowMessages: ["/cosmos.gov.v1.Vote"],
```

Reply:
If you want just the cosmos.gov.v1.Vote message to be allowed, we can change it to this. Can I proceed to set AllowMessages to ["/cosmos.gov.v1.Vote"]?

Reply:
I wonder about the vulnerability when we set this as a `*`. If it doesn't, `*` is okay.
And, I don't exactly know what governance features require us to use messages when operating interchain. I think we could leave it as a `*` for now.

Reply:
@koofree There should be no vulnerability in using `*` (at least in theory), since this is the configuration used by `gaia` and `juno` for example.
If you want to be more conservative, we could use the approach that is used by `osmosis`, which allows the following cosmos/ibc/wasm messages at the moment:
```
- /ibc.applications.transfer.v1.MsgTransfer

- /cosmos.bank.v1beta1.MsgSend

- /cosmos.staking.v1beta1.MsgDelegate
- /cosmos.staking.v1beta1.MsgBeginRedelegate
- /cosmos.staking.v1beta1.MsgCreateValidator
- /cosmos.staking.v1beta1.MsgEditValidator
- /cosmos.staking.v1beta1.MsgUndelegate

- /cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward
- /cosmos.distribution.v1beta1.MsgSetWithdrawAddress
- /cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission
- /cosmos.distribution.v1beta1.MsgFundCommunityPool

- /cosmos.gov.v1beta1.MsgVote

- /cosmwasm.wasm.v1.MsgExecuteContract
- /cosmwasm.wasm.v1.MsgInstantiateContract

- /cosmos.authz.v1beta1.MsgGrant
- /cosmos.authz.v1beta1.MsgRevoke
```

Reply:
Hi @niilptr! I clearly understood. If we can choose the conservative way, I have no doubt doing that way. Can we go on your suggestion?